tests/test_flow/test_reload.py

```diff
@@ -0,0 +1,258 @@
+                for f, r in expected_results_by_flow.items():
+                    _verify_flow_contents(f, r)
+
+        def reload_flow(self):
+            if reload_method == "reload":
+                self.flow.reload()
+            else:
+                self.flow = self.flow.reloading()
+
+        def get(self, name):
+            return self.flow.get(name)
+
+    def _flow_tester(name):
+        return FlowTester(name)
+
+    return _flow_tester
+
+
+def test_no_change(write_flow_module, flow_tester):
+    write_flow_module(1)
+    tester = flow_tester("flow")
+    tester.expect_values(x=1, x_doubled=2)
+
+    tester.reload_flow()
+    tester.expect_values(x=1, x_doubled=2)
+
+
+def test_change_entity_value(write_flow_module, flow_tester):
+    write_flow_module(1)
+    tester = flow_tester("flow")
+    tester.expect_values(x=1, x_doubled=2)
+
+    write_flow_module(2)
+    tester.reload_flow()
+    tester.expect_values(x=2, x_doubled=4)
+
+
+def test_change_entity_name(write_module, flow_tester):
+    def write_flow_module(z: int):
+        write_module(
+            "flow_module",
+            f"""
+                import bionic as bn
+
+                builder = bn.FlowBuilder('flow')
+                builder.assign('x', 1)
+
+                @builder
+                def x_plus_{z}(x):
+                    return x + {z}
+
+                flow = builder.build()
+            """,
+        )
+
+    write_flow_module(1)
+    tester = flow_tester("flow")
+    tester.expect_values(x=1, x_plus_1=2)
+
+    write_flow_module(2)
+    tester.reload_flow()
+    tester.expect_values(x=1, x_plus_2=3)
+    with pytest.raises(UndefinedEntityError):
+        tester.get("x_plus_1")
+
+
+def test_change_version(write_module, flow_tester):
+    def write_flow_module(x: int, y: int):
+        write_module(
+            "flow_module",
+            f"""
+                import bionic as bn
+
+                builder = bn.FlowBuilder('flow')
+
+                @builder
+                @bn.version(major={x}, minor={y})
+                def total():
+                    return {x} + {y}
+
+                flow = builder.build()
+            """,
+        )
+
+    write_flow_module(10, 1)
+    tester = flow_tester("flow")
+    tester.expect_values(total=11)
+
+    # Major version remains unchanged, cached value is returned
+    write_flow_module(10, 2)
+    tester.reload_flow()
+    tester.expect_values(total=11)
+
+    # Major version changed, a new result is computed
+    write_flow_module(20, 2)
+    tester.reload_flow()
+    tester.expect_values(total=22)
+
+
+def test_assist_versioning_mode(write_module, flow_tester):
+    def write_flow_module(x: int):
+        write_module(
+            "flow_module",
+            f"""
+                import bionic as bn
+
+                builder = bn.FlowBuilder('flow')
+                builder.set('core__versioning_mode', 'assist')
+
+                @builder
+                def total():
```

Comment:
Why is this called `total` and not, say, `x`?

Reply:
Renamed